src/thor/bidirectional_astar.cc

```diff
@@ -753,12 +773,20 @@ bool BidirectionalAStar::SetForwardConnection(GraphReader& graphreader, const BD
 
   // Keep the best ones at the front all others to the back
   best_connections_.emplace_back(CandidateConnection{pred.edgeid(), oppedge, c});
+
   if (c < best_connections_.front().cost)
     std::swap(best_connections_.front(), best_connections_.back());
 
-  // Set a threshold to extend search
-  if (threshold_ == std::numeric_limits<float>::max()) {
-    threshold_ = std::max(pred.sortcost() + cost_diff_, opp_pred.sortcost()) + kThresholdDelta;
+  // Set thresholds to extend search
+  if (cost_threshold_ == std::numeric_limits<float>::max()) {
+    float sortcost = std::max(pred.sortcost() + cost_diff_, opp_pred.sortcost());
+    if (desired_paths_count_ == 1) {
+      cost_threshold_ = sortcost + kThresholdDelta;
+    } else {
+      cost_threshold_ = sortcost + std::max(kAlternativeCostExtend * sortcost, kThresholdDelta);
+      iterations_threshold_ =
+          edgelabels_forward_.size() + edgelabels_reverse_.size() + kAlternativeIterationsDelta;
+    }
```

Comment:
seems like the coverage check is wrong, there are definitely unit tests that would have `desired_paths_count > 1`

Reply:
oh i see just randomly the other direction gets hit by the test instead of this one nevermind